### PR TITLE
create tests for AccountApiView

### PR DESCRIPTION
**Pull request by Lex-23**
First tests in the project (endpoint - AccountApiView)

### CODE REVIEW

File: cryptostock/account/tests/test_account_api_view.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    assert first_user_account.id == response["id"]
+
+
+def test_data_get_self_account(first_user_account, get_request1):
+    response = get_request1
+    assert first_user_account.id == response["id"]
+    assert first_user_account.owner.username == response["owner"]
+    assert first_user_account.name == response["name"]
+    assert first_user_account.wallet.id == response["wallet"]["id"]
+    assert first_user_account.cash_balance == 0
```
Comment (vadim-denisov) [inline cryptostock/account/tests/test_account_api_view.py:69]:
```suggestion
    assert response == {"id": first_user_account.id, ...}
```

Comment (vadim-denisov) [reply cryptostock/account/tests/test_account_api_view.py:69]: Your approach good as well. But usually we use following order in asserts:
`assert actual == expected`. You have reverse `assert expected == actual`

File: cryptostock/account/urls.py [hunk 2]
```diff
@@ -14,5 +14,5 @@
     path("salesdashboard/<int:pk>/buy/", NewOfferApiView.as_view()),
     path("offers/", OffersListApiView.as_view()),
     path("offers/<int:pk>/", OfferApiView.as_view()),
-    path("account/", AccountApiView.as_view()),
+    path("account/", AccountApiView.as_view(), name="self_account"),
```
Comment (vadim-denisov) [inline cryptostock/account/urls.py:17]: Don't need if we can pass to client full url `/api/account/`

File: cryptostock/cryptostock/settings.py [hunk 3]
```diff
@@ -26,7 +26,7 @@
 # SECURITY WARNING: don't run with debug turned on in production!
 DEBUG = True
 
-ALLOWED_HOSTS = []
+ALLOWED_HOSTS = ["testserver"]
```
Comment (vadim-denisov) [inline cryptostock/cryptostock/settings.py:29]: Why?

File: pyproject.toml [hunk 4]
```diff
@@ -10,6 +10,7 @@ Django = "^3.2.8"
 psycopg2-binary = "^2.9.1"
 djangorestframework = "^3.12.4"
 djangorestframework-simplejwt = "^5.0.0"
+pytest-django = "^4.4.0"
```
Comment (vadim-denisov) [inline pyproject.toml:13]: It's dev dependency

File: cryptostock/account/models.py [hunk 5]
```diff
@@ -13,7 +13,7 @@ class Account(models.Model):
     wallet = models.OneToOneField(
         Wallet, on_delete=models.CASCADE, related_name="account"
     )
-    cash_balance = CountField(default=0)
+    cash_balance = CountField()
```
Comment (vadim-denisov) [inline cryptostock/account/models.py:16]: Do we need migration for this chage?

File: cryptostock/account/tests/test_account_api_view.py [hunk 6]
```diff
@@ -0,0 +1,28 @@
+def test_get_self_account(first_user_account, auth_client1):
+    response = (auth_client1.get("/api/account/")).json()
+    assert response["id"] == first_user_account.id
```
Comment (vadim-denisov) [inline cryptostock/account/tests/test_account_api_view.py:3]:
```suggestion
    response = auth_client1.get("/api/account/")
    assert response.json() == {"id": first_user_account.id}
```

File: cryptostock/cryptostock/settings.py (hunk 3 above)
Comment (Lex-23) [reply cryptostock/cryptostock/settings.py:29]: it's for ApiClient()

File: cryptostock/account/tests/conftest.py [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    client = APIClient()
+    client.force_authenticate(user=first_user)
+    return client
+
+
+@pytest.fixture
+def auth_client2(second_user):
+    client = APIClient()
+    client.force_authenticate(user=second_user)
+    return client
```
Comment (vadim-denisov) [inline cryptostock/account/tests/conftest.py:53]:
```suggestion

@pytest.fixture
def api_client():
    return APIClient()
 
 
@pytest.fixture
def auth_first_user(api_client, first_user):
    api_client.force_authenticate(user=first_user)


@pytest.fixture
def auth_second_user(api_client, second_user):
    api_client.force_authenticate(user=second_user)
```

Comment (vadim-denisov) [reply cryptostock/account/tests/conftest.py:53]: In test
```
def test_x(api_client, auth_first_user):
    # api_client should be authentificated
```

File: cryptostock/account/tests/test_account_api_view.py [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+        "owner": first_user_account.owner.username,
+        "cash_balance": "0.0000",
+        "wallet": {
+            "id": first_user_account.wallet.id,
+            "name": first_user_account.wallet.name,
+        },
+        "wallet_records": [],
+    }
+    assert actual_data == response
+    assert 0 == first_user_account.cash_balance
```
Comment (vadim-denisov) [inline cryptostock/account/tests/test_account_api_view.py:20]:
```suggestion
    response = auth_client1.get("/api/account/")

    assert response.json() == {
        "id": first_user_account.id,
        "name": first_user_account.name,
        "owner": first_user_account.owner.username,
        "cash_balance": "0.0000",
        "wallet": {
            "id": first_user_account.wallet.id,
            "name": first_user_account.wallet.name,
        },
        "wallet_records": [],
    }
```

File: cryptostock/cryptostock/settings.py (hunk 3 above)
Comment (Lex-23) [reply cryptostock/cryptostock/settings.py]: only for test client in shell. I added this to dev.env

File: pyproject.toml (hunk 4 above)
Comment (Lex-23) [reply pyproject.toml]: done

File: cryptostock/account/tests/test_account_api_view.py (hunk 1 above)
Comment (vadim-denisov) [reply cryptostock/account/tests/test_account_api_view.py]: @Lex-23 Any questions on that?

File: cryptostock/account/models.py (hunk 5 above)
Comment (Lex-23) [reply cryptostock/account/models.py:16]: no - it was duplicate default

File: cryptostock/account/tests/test_account_api_view.py (hunk 1 above)
Comment (Lex-23) [reply cryptostock/account/tests/test_account_api_view.py]: Only question -  if I right understand this?

File: cryptostock/account/tests/test_account_api_view.py (hunk 6 above)
Comment (Lex-23) [reply cryptostock/account/tests/test_account_api_view.py]: done

File: cryptostock/account/tests/conftest.py (hunk 7 above)
Comment (Lex-23) [reply cryptostock/account/tests/conftest.py]: done

File: cryptostock/account/tests/test_account_api_view.py (hunk 1 above)
Comment (Lex-23) [reply cryptostock/account/tests/test_account_api_view.py]: I got it wrong right away

File: cryptostock/account/tests/test_account_api_view.py (hunk 8 above)
Comment (Lex-23) [reply cryptostock/account/tests/test_account_api_view.py]: done

File: cryptostock/account/tests/test_account_api_view.py [hunk 9]
```diff
@@ -0,0 +1,33 @@
+#  for all api test cases api_client should be authenticated
+
+
+def test_get_data_self_account(first_user_account, auth_first_user):
```
Comment (vadim-denisov) [inline cryptostock/account/tests/test_account_api_view.py:4]:
```suggestion
def test_get_account(first_user_account, auth_first_user):
```

File: cryptostock/account/tests/test_account_api_view.py [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+        "cash_balance": "0.0000",
+        "wallet": {
+            "id": first_user_account.wallet.id,
+            "name": first_user_account.wallet.name,
+        },
+        "wallet_records": [],
+    }
+
+
+def test_user_not_get_another_account(
```
Comment (vadim-denisov) [inline cryptostock/account/tests/test_account_api_view.py:19]:
```suggestion
def test_get_account_another_user(
```

**Comment by vadim-denisov** [comment]
Also add negative case test:
1) Get account if not authentificated

File: cryptostock/account/tests/test_account_api_view.py [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+    first_user_account, second_user_account, auth_second_user
+):
+    response = auth_second_user.get("/api/account/")
```
Comment (vadim-denisov) [inline cryptostock/account/tests/test_account_api_view.py:22]: Check status code as well.

Comment (Lex-23) [reply cryptostock/account/tests/test_account_api_view.py:23]: done

File: cryptostock/account/tests/test_account_api_view.py (hunk 10 above)
Comment (Lex-23) [reply cryptostock/account/tests/test_account_api_view.py]: done

File: cryptostock/account/tests/test_account_api_view.py (hunk 9 above)
Comment (Lex-23) [reply cryptostock/account/tests/test_account_api_view.py]: done

**Review by vadim-denisov — APPROVED** [review]

**Comment by Lex-23** [comment]
[write pytests](https://trello.com/c/KrUqCZfN/6-write-pytests)

**Pull request merged**